### PR TITLE
Update Azure's fault domains list

### PR DESCRIPTION
**Pull request by moelsayed**
**What this PR does / why we need it**:
Update the list of Azure Fault Domains per location. 
**Which issue(s) this PR fixes**:

Fixes #

**What type of PR is this?**

/kind bug
**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
Fix an issue with Azure support that prevented successful provisioning of user clusters on some Azure locations.
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: pkg/provider/cloud/azure/provider.go [hunk 1]
```diff
@@ -74,37 +75,79 @@ func New(dc *kubermaticv1.Datacenter, secretKeyGetter provider.SecretKeySelector
...
+		"uksouth":            2,
+		"ukwest":             2,
+		"westcentralus":      2,
+		"westeurope":         3,
+		"WestIndia":          2,
+		"westus":             3,
+		"westus2":            3,
+		"WestUS3":            3,
+	}
+	for region := range faultDomainsPerRegion {
```
Comment (xrstf) [inline pkg/provider/cloud/azure/provider.go]: You could simply `for region, count := range ...` and then avoid the 2nd slice access in line 147, no? It's fine for the loop to overwrite `count` on every iteration.

**Review by xrstf — APPROVED** [review]
/approve

**Comment by xrstf** [comment]
/override pre-kubermatic-dualstack-e2e-aws-canal
/override pre-kubermatic-dualstack-e2e-aws-cilium

(unrelated Rockylinux issue)

**Comment by embik** [comment]
@moelsayed can you please add a release note?

**Comment by embik** [comment]
We probably want to backport this as well.

**Pull request merged**